### PR TITLE
Copy a new instance of options dict for each child

### PR DESCRIPTION
**Pull request by adiosnb**
Closes https://github.com/teemtee/tmt/issues/1977

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -694,7 +694,14 @@ def __init__(self, **kwargs: Any) -> None:
             super().__init__(**kwargs)
 
 
-class Common(_CommonBase):
+class CopyOptionsMeta(type):
+    def __new__(cls: Any, name: str, bases: Tuple[type], dct: Dict[str, Any]) -> Any:
+        x = super().__new__(cls, name, bases, dct)
+        x._options = x._options.copy()
```
Comment (happz) [inline tmt/utils.py:700]: I'd copy also the `_context`, these two objects are tied together, `_options` are `_context.params`.

Comment (adiosnb) [reply tmt/utils.py:668]: The metaclass is used during child class inheritance by importlib, and the `_context` parameter is still the default `None` value, so there's nothing to copy.

**Comment by adiosnb** [comment]
It seems that this code causes the following error in python3.6.

```
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

File: tmt/utils.py (hunk 1 above)
Comment (happz) [reply tmt/utils.py:668]: I see, right, there's an initial empty dictionary. Could you add a comment here, that `_context` does not need to be copied? Once we start dealing with `_context` and `_options`, it'd be nice to be reminded of this fact :)

**Pull request closed without merging**